Assembus/Assets/Scripts/MainScreen/Sidebar/HierarchyView/HierarchyItemController.cs

```diff
@@ -492,87 +492,98 @@ public void ClickRelease(BaseEventData data)
-                {
-                    Icon = visible ? contextMenu.hide : contextMenu.show,
-                    Name = visible ? "Hide Item" : "Show Item",
-                    Action = () => ShowItem(!visible)
-                }
-            );
+            var entries = new List<ContextMenuController.Item>();
 
             var isGroup = itemInfo.ItemInfo.isGroup;
-            if (isGroup)
-                entries.Add(
-                    new ContextMenuController.Item
-                    {
-                        Icon = contextMenu.show,
-                        Name = "Show All",
-                        Action = ShowGroup
-                    }
-                );
-
-            if (hierarchyViewController.SelectedItems.Contains(this))
-                entries.Add(
-                    new ContextMenuController.Item
-                    {
-                        Icon = contextMenu.folder,
-                        Name = "Group Selected",
-                        Action = MoveToNewGroup
-                    }
-                );
-
-            if (isGroup)
+            if (!multiple)
             {
                 entries.Add(
                     new ContextMenuController.Item
                     {
-                        Icon = itemInfo.ItemInfo.isFused ? contextMenu.defuse : contextMenu.fuse,
-                        Name = itemInfo.ItemInfo.isFused ? "Split Group" : "Fuse Group",
-                        Action = FuseGroup
+                        Icon = contextMenu.edit,
+                        Name = "Rename",
+                        Action = () => RenameItem()
                     }
                 );
 
+                var visible = item.activeSelf;
                 entries.Add(
                     new ContextMenuController.Item
                     {
-                        Icon = contextMenu.add,
-                        Name = "Add Group",
-                        Action = AddGroup
+                        Icon = visible ? contextMenu.hide : contextMenu.show,
+                        Name = visible ? "Hide Item" : "Show Item",
+                        Action = () => ShowItem(!visible)
                     }
                 );
 
-                entries.Add(
-                    new ContextMenuController.Item
-                    {
-                        Icon = contextMenu.delete,
-                        Name = "Delete",
-                        Action = DeleteGroup
-                    }
-                );
+                if (isGroup)
+                    entries.Add(
+                        new ContextMenuController.Item
+                        {
+                            Icon = contextMenu.show,
+                            Name = "Show All",
+                            Action = ShowGroup
+                        }
+                    );
             }
 
-            if (stationController.IsOpen)
+            if (hierarchyViewController.SelectedItems.Contains(this))
```

Comment:
Additionaly check if the selection contains a assembly station might make sense here!?